### PR TITLE
Introduce `#[dynomite(skip_serializing_if = "..")]` attribute

### PR DESCRIPTION
**Pull request by Veetaha**
This basically reflects `serde::skip_serializing_if` attribute.
This will be very useful for us to get rid of manual workarounds where we need to store empty binary sets in a db record.

Also had to fix a clippy lint of implementing `Into` trait instead of `From` to let CI pass

cc @ZaMaZaN4iK @softprops

### CODE REVIEW

File: dynomite-derive/src/attr.rs [hunk 1]
```diff
@@ -26,18 +26,33 @@ pub(crate) type VariantAttr = Attr<VariantAttrKind>;
...
+
     /// Denotes Item sort key
     SortKey,
+
     /// Denotes a field that should be replaced with all of its subfields
     Flatten,
+
+    /// Denotes a field that should not be present in the resulting `Attributes` map
+    /// if the given function returns `true` for its value
+    SkipSerializingIf(Box<Expr>),
```
Comment (Veetaha) [inline dynomite-derive/src/attr.rs:44]: Had to use `Box` to make clippy happy

File: dynomite/tests/integration/main.rs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// With this `macro_use` we are able to use standard `assert_eq/assert_ne`
```
Comment (Veetaha) [inline dynomite/tests/integration/main.rs:1]: Moved this into a single integration tests as per [this article](https://matklad.github.io/2021/02/27/delete-cargo-integration-tests.html)

**Review by zamazan4ik — APPROVED** [review]
Thanks for the PR!

LGTM

**Comment by Veetaha** [comment]
Closing this, since the original maintainer asked for proceeding the work in the original repo, it's already merged there

**Pull request closed without merging**